Source/WebKit/WebProcess/WebPage/WebBackForwardListProxy.cpp

```diff
@@ -58,15 +58,28 @@ WebBackForwardListProxy::WebBackForwardListProxy(WebPage& page)
 {
 }
 
+static const unsigned DefaultCapacity = 100; // afryer: TODO: this should use the same const (which will improve readability).
+
 void WebBackForwardListProxy::addItem(Ref<HistoryItem>&& item)
 {
     RefPtr page = m_page.get();
     if (!page)
         return;
 
     LOG(BackForward, "(Back/Forward) WebProcess pid %i setting item %p for id %s with url %s", getCurrentProcessID(), item.ptr(), item->itemID().toString().utf8().data(), item->urlString().utf8().data());
-    m_cachedBackForwardListCounts = std::nullopt;
-    page->send(Messages::WebPageProxy::BackForwardAddItem(toFrameState(item.get())));
+    if (m_cachedBackForwardListCounts) {
+        if (!page->mainFrame() || page->mainFrame()->frameType() == Frame::FrameType::Local) {
```

Comment:
Why are Remote frames immune from this change? Are they handled in a different back/forward behavior? Or maybe there is no such thing as a non-Local frame that is also a mainFrame?